TombEngine/Objects/TR5/Entity/tr5_imp.cpp

```diff
@@ -96,206 +135,252 @@ namespace TEN::Entities::Creatures::TR5
+	{
+		if (!CreatureActive(itemNumber))
+			return;
+		
+		AI_INFO AI;
+		short angle = 0;
+		short otherAngle = 0;
+		short torso_y = 0;
+		short torso_x = 0;
+		short head_y = 0;
+		short head_x = 0;
 
-				if (creature->Enemy == LaraItem)
-					angle2 = AI.angle;
+		auto* item = &g_Level.Items[itemNumber];
+		auto* creature = GetCreatureInfo(item);
+
+		if (item->HitPoints <= 0)
+		{
+			if (item->Animation.ActiveState != IMP_STATE_DEATH)
+			{
+				CreatureAIInfo(item, &AI);
+				if (AI.angle > -IMP_SHOOT_DEATH_ANGLE_BACKWARD && AI.angle < IMP_SHOOT_DEATH_ANGLE_BACKWARD)
+					SetAnimation(item, IMP_ANIM_DEATH_BACKWARDS);
 				else
-					angle2 = phd_atan(LaraItem->Pose.Position.z - item->Pose.Position.z, LaraItem->Pose.Position.x - item->Pose.Position.x) - item->Pose.Orientation.y;
+					SetAnimation(item, IMP_ANIM_DEATH_FORWARD);
+			}
+		}
+		else
+		{
+			if (item->AIBits)
+				GetAITarget(creature);
+			else if (creature->HurtByLara)
+				creature->Enemy = LaraItem;
 
-				int d1 = item->Pose.Position.y - LaraItem->Pose.Position.y + CLICK(1.5f);
+			CreatureAIInfo(item, &AI);
 
-				if (LaraItem->Animation.ActiveState == LS_CROUCH_IDLE ||
-					LaraItem->Animation.ActiveState == LS_CROUCH_ROLL ||
-					LaraItem->Animation.ActiveState > LS_MONKEY_TURN_180 &&
-					LaraItem->Animation.ActiveState < LS_HANG_TO_CRAWL ||
-					LaraItem->Animation.ActiveState == LS_CROUCH_TURN_LEFT ||
-					LaraItem->Animation.ActiveState == LS_CROUCH_TURN_RIGHT)
+			int elevation = item->Pose.Position.y - creature->Enemy->Pose.Position.y + CLICK(1.5f);
+			if (creature->Enemy->IsLara())
+			{
+				otherAngle = AI.angle;
+				if (creature->Enemy->Animation.ActiveState == LS_CROUCH_IDLE ||
+					creature->Enemy->Animation.ActiveState == LS_CROUCH_ROLL ||
+					creature->Enemy->Animation.ActiveState > LS_MONKEY_TURN_180 &&
+					creature->Enemy->Animation.ActiveState < LS_HANG_TO_CRAWL ||
+					creature->Enemy->Animation.ActiveState == LS_CROUCH_TURN_LEFT ||
+					creature->Enemy->Animation.ActiveState == LS_CROUCH_TURN_RIGHT)
 				{
-					d1 = item->Pose.Position.y - LaraItem->Pose.Position.y;
+					elevation = item->Pose.Position.y - creature->Enemy->Pose.Position.y;
 				}
+			}
+			else
+				otherAngle = phd_atan(creature->Enemy->Pose.Position.z - item->Pose.Position.z, creature->Enemy->Pose.Position.x - item->Pose.Position.x) - item->Pose.Orientation.y;
+			
+			AI.xAngle = phd_atan(sqrt(AI.distance), elevation);
 
-				int d2 = sqrt(AI.distance);
-
-				AI.xAngle = phd_atan(d2, d1);
-
-				GetCreatureMood(item, &AI, true);
+			GetCreatureMood(item, &AI, true);
+			if (item->Animation.ActiveState == IMP_STATE_SCARED)
+				creature->Mood = MoodType::Escape;
+			CreatureMood(item, &AI, true);
 
-				if (item->Animation.ActiveState == IMP_STATE_SCARED)
-					creature->Mood = MoodType::Escape;
+			angle = CreatureTurn(item, creature->MaxTurn);
 
-				CreatureMood(item, &AI, true);
+			if (Wibble & 0x10)
+				item->SetMeshSwapFlags(0x400);
```

Comment:
This 0x400 can be demagic

0x400 hex, in decimal is 1024, but in binary is: 0000 0100 0000 0000

The mesh swaps use those binary values to indicate which mesh must be swap. In this case the 1 is in the 10th bit (starting by 0, 1, 2...). And if you check the Wad Tool, you'll see the 10th mesh is the head.

So it can be created a constant value at the top giving a name and the mesh number, to later use that name instead of the 0x400 that may be harder to identify.

If you want to see examples, Guide code is full of these.

So it could change it with:
```c++
/Adding the initialization at top with the other constants
const auto ImpHeadSwapJoints	 = std::vector<unsigned int>{ 10 };  

//in the SetMeshSwapFlags
item->SetMeshSwapFlags( ImpHeadSwapJoints );
```

It's using a vector, because it can keep more than 1 mesh. Imagine if the binary number were:
0000 0100 0010 0100
Then the swap mesh would change the 2nd, 5th and 10th meshes (left leg, body and head).

I hope this info helps.